channeldb/channel.go

```diff
@@ -2409,16 +2409,24 @@ func (c *OpenChannel) SetFwdFilter(height uint64, fwdFilter *PkgFilter) error {
 	})
 }
 
-// RemoveFwdPkg atomically removes a forwarding package specified by the remote
-// commitment height.
+// RemoveFwdPkgs atomically removes forwarding packages specified by the remote
+// commitment heights. If one of the intermediate RemovePkg calls fails, then the
+// later packages won't be removed.
 //
 // NOTE: This method should only be called on packages marked FwdStateCompleted.
-func (c *OpenChannel) RemoveFwdPkg(height uint64) error {
+func (c *OpenChannel) RemoveFwdPkgs(heights ...uint64) error {
 	c.Lock()
 	defer c.Unlock()
 
 	return kvdb.Update(c.Db, func(tx kvdb.RwTx) error {
-		return c.Packager.RemovePkg(tx, height)
+		for _, height := range heights {
```

Comment:
Are we sure that the bulk remove will actually reduce db contention? Not saying it's not better to do this in one tx, just asking if the assumption is measurable.

Reply:
I think it will be less noticeable when the ticker is fired once an hour. If we were to keep the ticker at one minute and batch them, there would be a much more noticeable performance improvement. Still I think it's better to have less db txn's as they can be quite expensive especially considering if we have many links doing these cleanup operations.